### PR TITLE
WIP trigonometric vectorized functions

### PR DESCRIPTION
**Pull request by miguelraz**
Very WIP... but a first draft was made!

### CODE REVIEW

File: crates/core_simd/src/ops.rs [hunk 1]
```diff
@@ -186,6 +186,30 @@ macro_rules! impl_op {
         }
     };
 
+    // trigonometric unary functions
+    { impl Sin for $type:ident, $scalar:ty } => {
+        impl_ref_ops! {
+            impl<const LANES: usize> core::ops::Sin for crate::$type<LANES> {
```
Comment (thomcc) [inline crates/core_simd/src/ops.rs:192]: Would we be adding these ops? I'd be personally against that — Not really our place and after a while I've come to believe the lack of core::num traits was the right call.

(If this was just a mistake caused by too much copypasta feel free to ignore)

Comment (miguelraz) [reply crates/core_simd/src/ops.rs:192]: Yes, I'm still grokking the codebase and am not sure how to add new functionality. I put up this PR to at least get the ball rolling. Recommendations on how to procceed are appreciated.

Comment (workingjubilee) [reply crates/core_simd/src/ops.rs:192]: For `fn sin` and `fn cos`, you want to have them directly be part of `impl SimdF32` and `impl SimdF64`, see my shuffle PR for something closer to how that can look. https://github.com/rust-lang/stdsimd/pull/62/files

**Comment by workingjubilee** [comment]
Raz does not know when he will be able to get back to this so closing.

**Comment by miguelraz** [comment]
Yup, bit off more than I can chew for the moment. Thanks for following up and others should feel free to take up the draft instead.

**Pull request closed without merging**